zkevm-circuits/src/circuit_tools/memory.rs

```diff
@@ -159,46 +202,69 @@ impl<F: Field, C: CellType> MemoryBank<F, C> {
         self.cur.expr()
     }
 
-    pub(crate) fn load(
-        &self,
-        description: &'static str,
-        cb: &mut ConstraintBuilder<F, C>,
-        offset: Expression<F>,
-        values: &[Expression<F>],
-    ) {
-        self.load_with_key(description, cb, self.key() - offset, values);
+    fn query_write(&mut self) -> Cell<F> {
+        let cell = self.writes.0[self.writes.1].clone();
+        self.writes.1 += 1;
```

Comment:
Have to think some more if this will work without problems or not, but I think it should. Similar style now as in the EVM which is nice (but EVM circuit uses the state circuit to check that accesses are correct logic).

Reply:
I think querying memory location like this is cute 😄 cuz it's just like `malloc()` in C++ that we query one memory location from the allocator.